[cmake] Don't repeat optional dependencies

fetzerch opened the pull request:
Commit fe50635fb7a09d1576890a5d548429d55af00221 (https://github.com/xbmc/xbmc/pull/10188) changed the behavior for optional dependencies. ON makes them a required dependency, AUTO an optional, OFF disables it. Therefore most of the defaults were incorrectly making most dependencies default.

This change removes explicit options for optional dependencies from CMakeLists.txt because they are implicitly added as AUTO in core_optional_dep and core_optional_dyload_dep.

@wsnipex: I guess the default should be AUTO on dev machines, right? And you just force it to ON when building packages.

This fixes the build on windows, where the avahi dependency doesn't exist.
Jenkins Builds:
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1516/ :x: 
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1517/ :x:
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1518/ :x: (unrelated issues)
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1519/ :x:
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1521/ :x:
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1523/ :x:

fetzerch committed:
[cmake] Fix options

Commit fe50635fb7a09d1576890a5d548429d55af00221 changed the behavior
of optional dependencies. ON now makes them a required dependency,
AUTO an optional one and OFF disables it. Options that were not listed
before were implicitly disabled (and supposed to be overridden in
platform specific files).

This change removes explicit options for optional dependencies from
CMakeLists.txt because they were now incorrectly made required
dependencies. In addition it adds options that default to off, because
they are now optional dependencies.

Paxxi commented:
Seems to work fine locally for me. Can't say if it breaks anything on other platforms though

fetzerch commented:
@FernetMenta: I think 767a74c got forgotten in the "drop deinterlace" change. Have put it here, because I want to get builds with cmake all green again.

FernetMenta commented:
@fetzerch I am not a fan of just build-fixing things. The change for IMX is not complete. IMO there is no benefit in pretending that IMX has adjusted to the changes while it actually misses some love. The changes for AML and OpenMax seem ok. Nobody cares about OpenMax anyway.

fetzerch commented:
@FernetMenta: Sry I wasn't aware of the state of IMX. Dropped the change for it.

wsnipex commented on project/cmake/CMakeLists.txt:
```diff
@@ -31,12 +24,9 @@ if(UNIX)
   option(ENABLE_OPENSSL     "Enable OpenSSL?" ON)
   option(ENABLE_SDL         "Enable SDL?" OFF)
-  option(ENABLE_X11         "Enable X11 support?" ON)
+  option(ENABLE_X11         "Enable X11 support?" OFF)
```
keep this ON please.

wsnipex commented on project/cmake/CMakeLists.txt:
```diff
@@ -194,20 +184,21 @@ if(NOT WIN32)
     else()
       core_require_dep(Sdl)
     endif()
+    core_optional_dep(EGL)
```
AFAIK EGL isn't optional anymore, right @FernetMenta @fritsch

FernetMenta replied:
correct

mapfau commented:
I really don't know what you guys have changed, but there are now some issues regarding [dorid] cmake builds:

1.) cmake claims not to find ALSA / AVAHI (I changed this by swithcing to OFF in CMakeLists.txt)
2.) There are compile issues regarding AML (wich fit into the topic here):

```
[ 96%] Building CXX object build/cores/VideoPlayer/videorenderers/hwdec/CMakeFiles/videorenderers_hwdec.dir/RendererAML.cpp.o
In file included from /home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp:21:0:
/home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.h:44:25: error: 'EDEINTERLACEMODE' has not been declared
   virtual bool Supports(EDEINTERLACEMODE mode);
                         ^
/home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp:84:29: error: 'bool CRendererAML::Supports' is not a static member of 'class CRendererAML'
 bool CRendererAML::Supports(EDEINTERLACEMODE mode)
                             ^
/home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp:84:29: error: 'EDEINTERLACEMODE' was not declared in this scope
/home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp:85:1: error: expected ',' or ';' before '{' token
 {
 ^  
```
AFAIR AML was compiled last time successfully, but questin remains, if AML should be build for [droid] because mediacodec is used by default, isn't it?

fetzerch committed:
[VideoPlayer] Fix build after dropping deinterlacing mode

25718f4ccebd227eaf63f0f8b53f0552f89f01e5 forgot to remove some methods
that still use EDEINTERLACEMODE.

fetzerch commented:
Builds are green on all platforms again: http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1528/
@wsnipex: good to go?
